src/java/org/apache/cassandra/utils/FBUtilities.java

```diff
@@ -130,16 +130,24 @@ public static MessageDigest newMessageDigest(String algorithm)
     public static InetAddress getJustLocalAddress()
     {
         if (localInetAddress == null)
-            try
-            {
-                localInetAddress = DatabaseDescriptor.getListenAddress() == null
-                                    ? InetAddress.getLocalHost()
-                                    : DatabaseDescriptor.getListenAddress();
-            }
-            catch (UnknownHostException e)
+        {
+            if (DatabaseDescriptor.getListenAddress() == null)
             {
-                throw new RuntimeException(e);
+                try
+                {
+                    localInetAddress = InetAddress.getLocalHost();
```

Comment:
```suggestion
                    localInetAddress = InetAddress.getLocalHost();
                    logger.info("InetAddress.getLocalHost() was used to resolve listen_address to {}, this can cause issues. Please check your node's config and set the listen_address in cassandra.yaml.", localInetAddress);
```

Reply:
@snazy I would reword

> InetAddress.getLocalHost() was used to resolve listen_address to {}, this can cause issues. Please check your node's config and set the listen_address in cassandra.yaml

to

> InetAddress.getLocalHost() was used to resolve listen_address to {}, double check this is correct. Please check your node's config and set the listen_address in cassandra.yaml accordingly if applicable.

Why? getLocalHost may be the right thing and `can cause issues` sounds like almost a certainty it will cause them. I'd favor a milder wording. Also listen_address you only set up unless you set a device to listen on iirc they are exclusive, so you set listen_address only if applicable.

Reply:
Yea, not opposed to a milder wording as long as it becomes clear that people should always configure it and not rely on (dangerous) defaults.